File: csharp/bob/Bob.cs
using System;
using System.Text.RegularExpressions;

public static class Bob
{
  public static string Response(string input)
  {
    var statement = input.Trim();

    if (String.IsNullOrWhiteSpace(statement))
      return "Fine. Be that way!";

    if (Regex.Matches(statement, @"[a-zA-Z]").Count > 0 &&
        statement.ToUpperInvariant() == statement)
    {
      return statement.EndsWith("?") ? 
        "Calm down, I know what I'm doing!" : 
        "Whoa, chill out!";
    }
    
    return statement.EndsWith("?") ? "Sure." : "Whatever.";
  }
}

File: java/bob/src/main/java/Bob.java
public class Bob {

  public String hey(String input) {

    if (input.trim().isEmpty())
      return "Fine. Be that way!";

    if (input.toUpperCase().equals(input) && 
        !input.matches("^[\\s+\\p{Punct}[0-9]]*$"))
      return "Whoa, chill out!";

    if (input.endsWith("?"))
      return "Sure.";

    return "Whatever.";
  }
}
